src/physics/circular_filament.rs

```diff
@@ -152,10 +310,269 @@ pub fn flux_density_circular_filament(
+///
+/// # Commentary
+///
+/// Near-exact formula (except numerically-evaluated elliptic integrals).
+/// The vector potential of a loop has zero r- and z- components due to symmetry,
+/// and does not vary in the phi-direction.
+///
+/// # References
+///
+///   \[1\] J. C. Simpson, J. E. Lane, C. D. Immer, R. C. Youngquist, and T. Steinrock,
+///         “Simple Analytic Expressions for the Magnetic Field of a Circular Current Loop,”
+///         Jan. 01, 2001. Accessed: Sep. 06, 2022. [Online]. Available: <https://ntrs.nasa.gov/citations/20010038494>
+pub fn vector_potential_circular_filament_par(
+    ifil: &[f64],
+    rfil: &[f64],
+    zfil: &[f64],
+    rprime: &[f64],
+    zprime: &[f64],
+    out_phi: &mut [f64],
+) -> Result<(), &'static str> {
+    // Chunk inputs
+    let ncores = std::thread::available_parallelism()
+        .unwrap_or(NonZeroUsize::MIN)
+        .get();
+
+    let n = (rprime.len() / ncores).max(1);
+
+    let rprimec = rprime.par_chunks(n);
+    let zprimec = zprime.par_chunks(n);
+
+    let outc = out_phi.par_chunks_mut(n);
+
+    // Run calcs
+    outc.zip(rprimec.zip(zprimec))
+        .try_for_each(|(outc, (rc, zc))| {
+            vector_potential_circular_filament(ifil, rfil, zfil, rc, zc, outc)
+        })?;
+
+    Ok(())
+}
+
+/// Off-axis A_phi component for a circular current filament in vacuum.
+///
+/// # Arguments
+///
+/// * `ifil`:    (A) current in each filament, length `m`
+/// * `rfil`:    (m) r-coord of each filament, length `m`
+/// * `zfil`:    (m) z-coord of each filament, length `m`
+/// * `rprime`:  (m) r-coord of each observation point, length `n`
+/// * `zprime`:  (m) z-coord of each observation point, length `n`
+/// * `out_phi`: (V-s/m), phi-component of magnetic vector potential at observation locations, length `n`
+///
+/// # Commentary
+///
+/// Near-exact formula (except numerically-evaluated elliptic integrals).
+/// The vector potential of a loop has zero r- and z- components due to symmetry,
+/// and does not vary in the phi-direction.
+///
+/// # References
+///
+///   \[1\] J. C. Simpson, J. E. Lane, C. D. Immer, R. C. Youngquist, and T. Steinrock,
+///         “Simple Analytic Expressions for the Magnetic Field of a Circular Current Loop,”
+///         Jan. 01, 2001. Accessed: Sep. 06, 2022. [Online]. Available: <https://ntrs.nasa.gov/citations/20010038494>
+pub fn vector_potential_circular_filament(
+    ifil: &[f64],
+    rfil: &[f64],
+    zfil: &[f64],
+    rprime: &[f64],
+    zprime: &[f64],
+    out_phi: &mut [f64],
+) -> Result<(), &'static str> {
+    let n = ifil.len();
+    let m = rprime.len();
+
+    // Check lengths; Error if they do not match
+    if rfil.len() != n || zfil.len() != n || zprime.len() != m || out_phi.len() != m {
+        return Err("Length mismatch");
+    }
+
+    for j in 0..m {
+        out_phi[j] = 0.0;
+    }
+
+    for i in 0..n {
+        for j in 0..m {
+            // Adjust from spherical coordinates
```

Comment:
The comment "Adjust from spherical coordinates" alone will be confusing to someone reading the comment who has not yet read the Simpson paper. I'd add a bit more context.
```suggestion
            // Eq. 1 and 2 of Simpson2001 give a formula for the vector potential of a loop in spherical coordinates.
            // Here, we use that formula adjusted to cylindrical coordinates.
```